pkg/nfd-worker/nfd-worker.go

```diff
@@ -140,15 +139,24 @@ func NewNfdWorker(args Args) (NfdWorker, error) {
 		&local.Source{},
 	}
 
-	sourceWhiteList := map[string]struct{}{}
-	for _, s := range args.Sources {
-		sourceWhiteList[strings.TrimSpace(s)] = struct{}{}
-	}
+	// Determine enabled feature
+	if len(args.Sources) == 1 && args.Sources[0] == "all" {
+		nfd.sources = allSources
+	} else {
+		// Add fake source which is only meant for testing. It will be enabled
+		// only if listed explicitly.
+		allSources = append(allSources, &fake.Source{})
 
-	nfd.sources = []source.FeatureSource{}
-	for _, s := range allSources {
-		if _, enabled := sourceWhiteList[s.Name()]; enabled {
-			nfd.sources = append(nfd.sources, s)
+		sourceWhiteList := map[string]struct{}{}
+		for _, s := range args.Sources {
+			sourceWhiteList[strings.TrimSpace(s)] = struct{}{}
+		}
+
+		nfd.sources = []source.FeatureSource{}
+		for _, s := range allSources {
+			if _, enabled := sourceWhiteList[s.Name()]; enabled {
+				nfd.sources = append(nfd.sources, s)
+			}
```

Comment:
Should we print a warning if the user specify a source name that does not match to any source?

Reply:
Mm, reasonable improvement. Fixed